src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIMarsManager.cs

```diff
@@ -50,5 +49,13 @@ public SNIMarsConnection GetConnection(SNIHandle lowerHandle)
         {
             return _connections[lowerHandle];
         }
+
+        /// <summary>
+        /// Clears all Mars connections managed by this class.
+        /// </summary>
+        public void ClearMarsConnections()
+        {
+            _connections.Clear();
```

Comment:
Is clearing it sufficient?  Do we need to dispose of the connections somehow to tear them down?  And if so, is it possible we'd attempt to dispose of them while they're still in use?